content/usgs-packages/geoknife_Execute.Rmd

```diff
@@ -0,0 +1,147 @@
+
+1. You are manually executing a job and manually checking it.
+2. You are running a script that kicks off a `geoknife` process followed by lines of code that use the returned data. 
+3. You are running a long `geoknife` process and want to be able to close R/RStudio and come back to a completed job later. 
+
+For the first scenario, the workflow from above was fine. If you are manually checking that the job has completed before trying to extract results, then nothing should fail.
+
+For the second scenario, your code will fail because it will continue to execute the code line by line after starting the job. So, your code will fail at the code that gets the data (`result`/`download`) since the job is still running. You can prevent scripts from continuing until the job is complete by using the function `wait`. This function makes a call to GDP at specified intervals to see if the job is complete, and allows the code to continue once the job is complete. This function has two arguments: the `geojob` object and `sleep.time`. `sleep.time` defines the interval at which to check the status of the job. Please try to adjust `sleep.time` to limit the number of calls to GDP, e.g. if you know the job will take about an hour, set `sleep.time=1800` (a half hour). The default for `sleep.time` is 5 seconds.
+
+```{r wait-workflow, message=FALSE, warning=FALSE}
+# typical wait workflow
+evap_geojob <- geoknife(evap_stencil, evap_fabric, evap_knife)
+wait(evap_geojob, sleep.time = 10)
+evap_data <- result(evap_geojob)
+```
+
+If you know ahead of time that your process will be long, you can tell the job to wait to continue when defining your knife (the default is to not wait). `sleep.time` can be specified as an argument to `webprocess`. The following is functionally the same as the use of `wait()` from above.
+
+```{r wait-shortcut, message=FALSE, warning=FALSE}
+# create knife with the args wait and sleep.time
+evap_knife <- webprocess(wait=TRUE, sleep.time=10)
+
+# follow the same code from before to get the unweighted algorithm
+all_algorithms <- query(evap_knife, 'algorithms')
+unw_algorithm <- all_algorithms[grep('unweighted', names(all_algorithms))]
+algorithm(evap_knife) <- unw_algorithm
+
+# create geojob using knife w/ wait
+evap_geojob <- geoknife(evap_stencil, evap_fabric, evap_knife)
+
+# get result
+evap_data <- result(evap_geojob)
+```
+
+As in the third scenario, if you have a job that will take a long time and plan to close R in the interim, you can specify the argument `email` when creating the knife. Then when you use your new knife in the `geoknife` call, it will send an email with appropriate information upon job completion. The email alert will contain the completed job URL and ID needed to pull down the data later using `result` (needs URL) or `download` (needs ID). 
```

Comment:
Can you describe what those new ways are? This won't be released until FY18, so it would probably be worth changing to the newer version methods